.github/workflows/container-nexus-publish.yml

```diff
@@ -0,0 +1,58 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright Red Hat Inc. and Hibernate Authors
+
+name: Push Hibernate Nexus3 Proxy to Quay.io
+
+on:
+  push:
+    branches:
+      - 'main'
+    paths:
+      - containers/nexus/**
```

Comment:
I've added this GH action to build and push to Quay. My thinking here is that we wouldn't want to associate tags in this repo to nexus image builds, or create a new docker tag for each merged commit ... so relying on quay build triggers doesn't look like an option. Instead, in the action, we can limit its runs to when the build-related files change.
I've tested it a bit on my fork here: https://github.com/marko-bekhta/hibernate-infra/actions/runs/13012497027/job/36293369488 and https://quay.io/repository/rh-ee-mbekhta/proxy-nexus3?tab=tags 

so the question would be ... do we want to do this through a GH action (or on CI) since we'd need to add a secret for Quay somewhere for the build? And, more importantly, 😄 how do we name it?

Reply:
I would personally prefer to keep all release stuff within ci.hibernate.org... but let's ask @beikov .

Reply:
That being said:

1. I'm not sure Quay.io creates a new docker tag for each merged commit... if the dockerfile or involved content results in identical layers, they would have identitical SHAs and thus no tag would need to be updated.
2. If it's simpler, we could just have one git repo per image. Or have a common repo for all "infra" images, and share the "version" (tag) amonst them -- creating new tags for all images even if only one changes. It's not like we have tons of images anyway.

Reply:
> I'm not sure Quay.io creates a new docker tag for each merged commit... 

Me neither, but ... when I was trying to test the GH trigger .... 

![image](https://github.com/user-attachments/assets/59e34ceb-0fb1-43af-841b-963293de4849)

the app wants access to waaay to many things (I'd say). So I never got past that request screen and haven't tested it further 😄.

and since we have `docker.withRegistry .... `  the build/publish might be as simple as:

https://github.com/hibernate/hibernate-infra/blob/ddfe7bb87d8227f5b7dbf242f914e23b546a4cac/ci/release/container/nexus/Jenkinsfile#L46-L60

Reply:
We already have this build trigger thing set up for https://quay.io/repository/hibernate/awestruct-build-env?tab=builds (see bottom of the page).

What's this app about? Because we have a "Quay" oauth app active in the Hibernate org on GitHub, but it's just an oauth app, so really only about authentication, I don't think it has access to much beyond that?

Reply:
> What's this app about? Because we have a "Quay" oauth app active in the Hibernate org on GitHub, but it's just an oauth app, so really only about authentication, I don't think it has access to much beyond that?

Okay, looks like oauth apps are more about "impersonating" members and thus accessing the resources members can access.

Anyway... like I said, it's already enabled. I'd tend to trust Quay requests access to resources it needs and uses that access reasonably and safely, but then maybe I'm biased, Quay being sponsored by Red Hat and all :)